zproject/email_backends.py

```diff
@@ -108,3 +110,46 @@ def send_messages(self, email_messages: Sequence[EmailMessage]) -> int:
                 email_log_url = settings.ROOT_DOMAIN_URI + "/emails"
                 logging.info("Emails sent in development are available at %s", email_log_url)
         return num_sent
+
+
+class PersistentSMTPEmailBackend(EmailBackend):
+    def _open(self, **kwargs: Any) -> bool | None:
+        is_opened = super().open()
+        if is_opened:
+            self.opened_at = timezone_now()
+            return True
+
+        return is_opened
+
+    @override
+    def open(self, **kwargs: Any) -> bool | None:
+        is_opened = self._open()
+        if is_opened:
+            return True
+
+        status = None
+        time_elapsed = (timezone_now() - self.opened_at).seconds / 60
+        if settings.EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES != 0 and (
```

Comment:
Through your current logic of returning early if `is_opened` is True, the `!= 0` check here is purely defensive, right?

This has been enough back-and-forth over this logic that it's very definitely going to need a comment in it working through why the logic is correct. :)